providers/jdbc/shedlock-provider-jdbc-template/src/main/java/net/javacrumbs/shedlock/provider/jdbctemplate/JdbcTemplateStorageAccessor.java

```diff
@@ -31,58 +35,72 @@
-	private final String tableName;
-	private final JdbcTemplate jdbcTemplate;
+    private final String tableName;
+    private final JdbcTemplate jdbcTemplate;
+    private final TransactionTemplate transactionTemplate;
 
-	JdbcTemplateStorageAccessor(JdbcTemplate jdbcTemplate, String tableName) {
-		this.jdbcTemplate = requireNonNull(jdbcTemplate, "jdbcTemplate can not be null");
-		this.tableName = requireNonNull(tableName, "tableName can not be null");
-	}
+    public JdbcTemplateStorageAccessor(JdbcTemplate jdbcTemplate, PlatformTransactionManager transactionManager, String tableName) {
+        this.jdbcTemplate = requireNonNull(jdbcTemplate, "jdbcTemplate can not be null");
+        this.tableName = requireNonNull(tableName, "tableName can not be null");
+        if (null == transactionManager) {
+            transactionManager = new DataSourceTransactionManager(jdbcTemplate.getDataSource());
+        }
+        this.transactionTemplate = new TransactionTemplate(transactionManager);
+        this.transactionTemplate.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
+    }
 
-	@Override
-	@Transactional(propagation = Propagation.REQUIRES_NEW)
-	public boolean insertRecord(LockConfiguration lockConfiguration) {
-		String sql = "INSERT INTO " + tableName + "(name, lock_until, locked_at, locked_by) VALUES(?, ?, ?, ?)";
-		try
-		{
-			int insertedRows = jdbcTemplate.update(sql, preparedStatement -> {
-				preparedStatement.setString(1, lockConfiguration.getName());
-				preparedStatement.setTimestamp(2, Timestamp.from(lockConfiguration.getLockAtMostUntil()));
-				preparedStatement.setTimestamp(3, Timestamp.from(Instant.now()));
-				preparedStatement.setString(4, getHostname());
-			});
-			return insertedRows > 0;
-		}
-		catch (DataIntegrityViolationException e)
-		{
-			return false;
-		}
-	}
+    JdbcTemplateStorageAccessor(JdbcTemplate jdbcTemplate, String tableName) {
+        this(jdbcTemplate, null, tableName);
+    }
 
-	@Override
-	@Transactional(propagation = Propagation.REQUIRES_NEW)
-	public boolean updateRecord(LockConfiguration lockConfiguration) {
-		String sql = "UPDATE " + tableName
-		             + " SET lock_until = ?, locked_at = ?, locked_by = ? WHERE name = ? AND lock_until <= ?";
-		int updatedRows = jdbcTemplate.update(sql, statement -> {
-			Timestamp now = Timestamp.from(Instant.now());
-			statement.setTimestamp(1, Timestamp.from(lockConfiguration.getLockAtMostUntil()));
-			statement.setTimestamp(2, now);
-			statement.setString(3, getHostname());
-			statement.setString(4, lockConfiguration.getName());
-			statement.setTimestamp(5, now);
-		});
-		return updatedRows > 0;
-	}
+    @Override
+    public boolean insertRecord(LockConfiguration lockConfiguration) {
+        String sql = "INSERT INTO " + tableName + "(name, lock_until, locked_at, locked_by) VALUES(?, ?, ?, ?)";
+        return transactionTemplate.execute(status -> {
```

Comment:
I am really sorry, I have mislead you. I am afraid that TransactionTemplate does not create a new transaction. What exactly is the problem with JTA and autocommit?

Reply:
TransactionTemplate creates a new transaction if the correct propagationBehavior is st (PROPAGATION_REQUIRES_NEW or PROPAGATION_REQUIRED), so this should work.
The problem with autocommit is that getConnection() possibly returns a connection with a transaction already started - this might commit an already running transaction (just a gut feeling, I could not verify this. But using TransactionTemplate is the clean solution I think.